### PR TITLE
chore(halo2):Pin halo2 branch to commit + Fix Halo2 KZG bindings

### PR DESCRIPTION
**Pull request by PatStiles**
Pins Halo2 bindings to stable commit

### CODE REVIEW

File: operator/halo2ipa/lib/libhalo2ipa_verifier.a
Comment (entropidelic) [inline operator/halo2ipa/lib/libhalo2ipa_verifier.a:1]: the binaries should be ignored

File: operator/halo2ipa/lib/libhalo2ipa_verifier.dylib
Comment (entropidelic) [inline operator/halo2ipa/lib/libhalo2ipa_verifier.dylib:1]: the binaries should be ignored

File: operator/halo2kzg/lib/libhalo2kzg_verifier.a
Comment (entropidelic) [inline operator/halo2kzg/lib/libhalo2kzg_verifier.a:1]: the binaries should be ignored

File: operator/halo2kzg/lib/libhalo2kzg_verifier.dylib
Comment (entropidelic) [inline operator/halo2kzg/lib/libhalo2kzg_verifier.dylib:1]: the binaries should be ignored

**Comment by entropidelic** [comment]
lgtm ~~just add the generated FFI libraries to the `.gitignore`~~

File: operator/halo2kzg/.gitignore [hunk 1]
```diff
@@ -1,3 +0,0 @@
-lib/libhalo2kzg_verifier.so
-lib/libhalo2kzg_verifier.a
-lib/libhalo2kzg_verifier.dylib
```
Comment (entropidelic) [inline operator/halo2kzg/.gitignore]: these shouldn't be deleted

**Review by entropidelic — APPROVED** [review]

**Review by NicolasRampoldi — APPROVED** [review]

**Pull request merged**